utils/src/utils/isDeepEqual.ts

```diff
@@ -0,0 +1,30 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+export function isDeepEqual(obj1: object, obj2: object): boolean {
```

Comment:
Currently this is only used by Container Apps to compare CA objects to check if there are unsaved changes.

There are many nuances to a correct deep equal function, and I'm worried we'll try to use this function somewhere else and it'll cause bugs. Could we instead either keep this specific to Container Apps and name is as such, OR use a library like https://github.com/chaijs/deep-eql that is a near perfect deep equal implementation? Ex: current implementation breaks if symbols are present as keys on the object.

Also, this is something that needs tests so we know what cases it does work for and what cases it doesn't work for.

Reply:
Agreed. I think we should expose it but I like the idea of using an underlying library.